src/modules/logger/logger.cpp

```diff
@@ -2146,6 +2148,10 @@ void Logger::write_version(LogType type)
 		write_info(type, "sys_os_ver", os_version);
 	}
 
+	if (git_tag && git_tag[0]) {
+		write_info(type, "ver_sw_tag", git_tag);
```

Comment:
I don't see why we need the tag, as we already have it through `px4_firmware_version()`, https://github.com/PX4/Firmware/pull/11676/files#diff-64b21add574f1f3f059c2abc18eae7d6R2121.

Reply:
As far as I understand, the `px4_firmware_version()` does not, in fact, return a tag, but rather [a version number](https://github.com/PX4/Firmware/blob/master/src/lib/version/version.c#L149-L152). That's fine for mainline firmwares, but we have modified firmware builds with additional information in the tag (our tags look like `v1.8.2-clever.5`). We'd like to see the whole tag to not guess which exact firmware we're using.

Reply:
We have custom version tags for this: https://dev.px4.io/en/setup/building_px4.html#firmware_version. Does that work for you?

Reply:
That's an interesting feature, but unfortunately I don't think it does. We use different firmwares for different models of drones, and it'd be more convenient to just use the full tag name (which contains the target drone name) instead of trying to differentiate by numbers.

Reply:
Having different firmwares for different models does not sound great either. Why not distinguish by using different SYS_AUTOSTART id's?

Reply:
@bkueng It's very useful when we have separate teams working on independent features. The tag helps you and your colleagues to identify the source branch.

Reply:
The branch name is already in the log, did you see that?

Reply:
We usually look through the log after the flight. Git tag string is important before the flight, e.g. during the manual vehicle inspection.

Reply:
That does not give me a reason why the branch is not enough.
What I want to avoid here is to have redundant information in different formats in the log. That is very confusing for anyone looking at the data.

Reply:
The possibility of getting tag and branch name BEFORE flight as a pre-flight check would be very useful because we often have to test different features (e. g. precision landing and different camera modules) on one prototype at a close periods of time.

Reply:
> The possibility of getting tag and branch name BEFORE flight as a pre-flight check would be very useful because we often have to test different features (e. g. precision landing and different camera modules) on one prototype at a close periods of time.

Which you do with `ver all`.